plasticos_facility_profile/models/facility_profile.py

```
@@ -30,47 +30,174 @@ class PlastosFacilityProfile(models.Model):
             if not rec.partner_id.parent_id:
-                raise ValidationError("Capability profile can only be attached to facility-level partners.")
+                raise ValidationError(
+                    "Capability profile can only be attached to "
+                    "facility-level partners."
+                )
+
+    @api.constrains("density_min", "density_max")
+    def _check_density_range(self):
+        for rec in self:
+            if rec.density_min and rec.density_max and rec.density_min > rec.density_max:
+                raise ValidationError("Density min cannot exceed density max.")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)